### DISCUSSION TITLE
How to create Subfolders

### DISCUSSION THREAD

**Discussion by marexx** [posthtml/posthtml#378] [Q&A] [answered]
Hi there,
i'm pretty new to NPM and posthtml.. 
i set up a little site which is rendering fine. But i would need to have subfolders in my html structure. How can i configure that?

this is my posthtml.json
{
    "input": "src/views/*.html",
    "output": "public",
    "plugins": {
        "posthtml-modules": {
            "root": "./src/views",
            "initial": true
        },
        "htmlnano": {}
    }
}

this is my call for posthtml:
'"build:html": "posthtml -c posthtml.json",`

this is how my src folder structure looks like:

src/views/components
head.html
footer.html
...

src/views/
index.html

src/views/subpage
index.html

what i would like to have in the end is
http://localhost:3000/supage/index.html
http://localhost:3000/index.html
...

**Comment by marexx** [comment]
when i use this call 
`"build:html": "posthtml src/views/**/*.html -o public -a",`
i get an error:
 Cannot destructure property 'config' of 'explorer.search(...)' as it is null.

**Comment by Scrum** [comment, marked as answer]
@marexx Hello, you need to put in `posthtml.json` `"input": "src/views/**/*.html"`

**Reply by marexx** [reply]
i tried this. but then all files were put in one folder, it doesn't create the subfolder structure..

**Reply by Scrum** [reply]
yes, because you need flug `allInOutput`

```json
{
    "root": "src/views",
    "input": "**/*.html",
    "output": "public",
    "allInOutput": true
}
```

**Reply by Scrum** [reply]
[example](https://github.com/inside-demo/posthtml-subfolder)

**Reply by marexx** [reply]
works, thanx a lot

**Reply by Scrum** [reply]
If the issue is resolved, don't forget to mark as answered, thanks.

**Comment by marexx** [comment]
i don't find the button to resolve it??

**Reply by Scrum** [reply]
<img width="925" alt="Screenshot 2022-06-23 at 11 53 51" src="https://user-images.githubusercontent.com/2789192/175259291-613bea1f-ef3a-49b8-8416-dc0af0c2b944.png">

**Reply by Scrum** [reply]
ohh, looks like you marked your comment as the answer, could you correct and mark my answer as the solution?

**Reply by marexx** [reply]
i can't mark yours as answer, the button only shows on my Comment??

**Reply by Scrum** [reply]
could you attach a screenshot?, i will forward it to support
